addons/sales_team/models/crm_team.py

```diff
@@ -62,6 +62,8 @@ def _get_default_team_id(self, user_id=None, domain=None):
         ])
         if teams and domain:
             filtered_teams = teams.filtered_domain(domain)
+            # Prefer the current company if possible
+            filtered_teams = filtered_teams.sorted(lambda team: not team.company_id or team.company_id == self.env.company, reverse=True)
```

Comment:
```suggestion
            filtered_teams = filtered_teams.sorted(lambda team: team.company_id == self.env.company, reverse=True)
```
team.company_id and self.env.company will always be comparable, no reason to check truthiness

Reply:
Yes there is. To ensure correct ordering, we need teams with company_id == False to also give true on this check. Otherwise, the case in the test file (Team1 belongs to Company1, Team2 belongs to Company2, Team3 has no company_id but the lowest sequence) would return the compatible companies in the wrong order.

Reply:
Ah right